### PR TITLE
refactor: 💡 Migrating to `Hds::Table` in OIDC auth-method form

### PR DESCRIPTION
**Pull request by dvonanderson**
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-11719

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-11719)

## Description
- Migrating field tables to `Hds::Table` from `Rose::Table` in OIDC auth method form.
- The `Form::Field::ListWrapper` doesn't support a `select` yield, so we are creating the table based on some patterns that are in used in the `form/auth-method/ldap` [template](https://github.com/hashicorp/boundary-ui/blob/8ad4e93ff7ccba6c25423089a7648e1f659aa6d6/ui/admin/app/components/form/auth-method/ldap/index.hbs#L385)
- Migrated `Rose::Buttons` to `Hds::Buttons`

### Before - OIDC new form:
image

### After - OIDC new form:
image

### Before - OIDC detail view:
image

### After - OIDC detail view:
Screenshot 2024-01-12 at 10 16 08

## How to Test
- Click on Vercel link for Admin UI or pull down PR and run it locally using `yarn start`
- Click on "Auth Method" in the sidebar navigation.
- Click on the "New" button in the top right hand corner of page and select OIDC.
- Ensure in the "Signing Algorithms" section you are able to select the different algorithms and add them to the table.
- Ensure in the "Account Claim Maps" section you are able to input and select the key-value pairs and add them to the table.
- After saving the form ensure that you are able to go back to the created OIDC and see that data in a disabled state.

:technologist: [Admin preview](PATH_TO_FEATURE)

## Checklist:

- [x] I have added before and after screenshots for UI changes
~~- [ ] I have added JSON response output for API changes~~
~~- [ ] I have added steps to reproduce and test for bug fixes in the description~~
~~- [ ] I have commented on my code, particularly in hard-to-understand areas~~
- [x] My changes generate no new warnings
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~

### CODE REVIEW

**Review by cameronperera — APPROVED** [review]
LGTM

**Review by lisbet-alvarez — APPROVED** [review]
LGTM

File: ui/admin/app/components/form/auth-method/oidc/index.hbs [hunk 1]
```diff
@@ -144,115 +144,116 @@
...
-            <row.headerCell>{{t 'titles.value'}}</row.headerCell>
-            <row.headerCell>{{t 'titles.actions'}}</row.headerCell>
-          </header.row>
-        </table.header>
-        <table.body as |body|>
-          {{#each @model.signing_algorithms as |algorithm i|}}
-            <body.row as |row|>
-              <row.cell>
 
+  <Hds::Form::Fieldset as |F|>
```
Comment (ZedLi) [inline ui/admin/app/components/form/auth-method/oidc/index.hbs:148]: This should be converted into a new sub component for the `list-wrapper` to support selects

**Review by ZedLi — CHANGES_REQUESTED** [review]

File: ui/admin/app/components/form/auth-method/oidc/index.hbs (hunk 1 above)
Comment (ZedLi) [reply ui/admin/app/components/form/auth-method/oidc/index.hbs:148]: Yeah, you can open a new PR for it and then use it in this one

**Comment by dvonanderson** [comment]
This PR will be handled in this [PR](https://github.com/hashicorp/boundary-ui/pull/2112)

**Pull request closed without merging**